cmd/earthly/main.go

```diff
@@ -2791,21 +2789,17 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 					return
 				}
 
-				id, err := cc.UploadLog(logPath)
-				if err != nil {
-					err := errors.Wrapf(err, "failed to upload log")
-					app.console.Warnf(err.Error())
-					return
-				}
+				var id string
+				id, uploadErr = cc.UploadLog(logPath)
 				app.console.Printf("Share your build log with this link: %s\n", id)
 			}()
 		} else {
 			// If you are not logged in, then advertise the service, since they probably turned it on to try it.
 			defer func() { // Defer this to keep log upload code together
-				if errors.Is(whoAmIErr, cloud.ErrUnauthorized) {
-					app.console.Printf("Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.")
+				if err != nil && !errors.Is(uploadErr, cloud.ErrUnauthorized) {
```

Comment:
is `err` intentional?
 
```suggestion
				if uploadErr != nil && !errors.Is(uploadErr, cloud.ErrUnauthorized) {
```

Reply:
Good eye. I refactored it last minute and make a mistake